synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	version, err := client.Version()
+	assert.NoError(t, err)
+	assert.Equal(t, "v3", version.APIVersion)
+	assert.NotEmpty(t, version.Version)
+}
+
+func TestIntegration_Config(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	config, err := client.Config()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, config)
+}
+
+func TestIntegration_Plugins(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugins, err := client.Plugins()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	for _, plugin := range plugins {
+		assert.NotEmpty(t, plugin.Name)
+		assert.NotEmpty(t, plugin.Maintainer)
+		assert.NotEmpty(t, plugin.Tag)
+		assert.NotEmpty(t, plugin.Description)
+		assert.NotEmpty(t, plugin.ID)
+		assert.True(t, plugin.Active)
+	}
+}
+
+func TestIntegration_PluginInfo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugins, err := client.Plugins()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	for _, p := range plugins {
+		plugin, err := client.Plugin(p.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, plugin.Name)
+		assert.NotEmpty(t, plugin.Maintainer)
+		assert.NotEmpty(t, plugin.Tag)
+		assert.NotEmpty(t, plugin.Description)
+		assert.NotEmpty(t, plugin.VCS)
+		assert.NotEmpty(t, plugin.ID)
+		assert.True(t, plugin.Active)
+		assert.NotEmpty(t, plugin.Network.Address)
+		assert.NotEmpty(t, plugin.Network.Protocol)
+		assert.NotEmpty(t, plugin.Version.PluginVersion)
+		assert.NotEmpty(t, plugin.Version.SDKVersion)
+		assert.NotEmpty(t, plugin.Version.BuildDate)
+		assert.NotEmpty(t, plugin.Version.GitCommit)
+		assert.NotEmpty(t, plugin.Version.GitTag)
+		assert.NotEmpty(t, plugin.Version.Arch)
+		assert.NotEmpty(t, plugin.Version.OS)
+		assert.NotEmpty(t, plugin.Health.Timestamp)
+		assert.Equal(t, "OK", plugin.Health.Status)
+		assert.Equal(t, 2, len(plugin.Health.Checks))
+
+		for _, check := range plugin.Health.Checks {
+			assert.NotEmpty(t, check.Name)
+			assert.Equal(t, "OK", check.Status)
+			assert.NotEmpty(t, check.Type)
+			assert.NotEmpty(t, check.Timestamp)
+			assert.Empty(t, check.Message)
+		}
+
+	}
+}
+
+func TestIntegration_PluginHealth(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	health, err := client.PluginHealth()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, health.Status)
```

Comment:
does this need to be a NotEmpty check? If we can check for `len(health.Healthy)` and `health.Active`, etc, then I imagine the health status should also be populated with a known value?